### ISSUE TITLE
Not working with iOS 17.1+

### ISSUE THREAD

**Issue by Arrankar** [slackhq/PanModal#202]
### Description

Describe your issue here.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x ] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [ x] I've read and understood the [Contributing guidelines](https://github.com/slackhq/PanModal/blob/master/CONTRIBUTING.md) and have done my best effort to follow them.
* [ x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [ x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Modal controller don't present on iOS 17.1 and upper

#### Reproducible in:

PanModal version: 1.2.7

iOS version: 17.2 beta

#### Steps to reproduce:

1. Create controller for presenting by PanModal
2. Present that controller by "presentPanModal" func
3. Look at dark screen without your controller.

#### Expected result:

Modal controller under dark screen.

#### Actual result:

Dark screen without modal controller.

#### Attachments:
<img width="328" alt="Снимок экрана 2023-10-31 в 00 18 58" src="https://github.com/slackhq/PanModal/assets/38263241/2b55603f-671c-4d57-89b6-ffb0c6a3c230">

**Comment by Noah-JJ** [comment]
What should we do?

**Comment by Arrankar** [comment]
> What should we do?

Start to search new library I guess

**Comment by yavl** [comment]
Yeah, I can also reproduce this on iOS 17. The strange thing was that some ViewControllers were showing up and some were not. Then I realized that the view controller won't show up on `presentPanModal` if you add subviews / set up auto layout constraints in `init()`, so in my case the fix was just to move addSubview, constraints code to `override func viewDidLoad()`.

No need to search new library 😃

**Comment by PhanithNY** [comment]
If you embedded viewController inside UINavigationController, try below code:
``` swift
    let modalVC = SheetViewController()
    let navigationController = UINavigationController()
    //    This isn't work
    //    let navigationController = UINavigationController(rootViewController: modalVC)
    navigationController.setViewControllers([modalVC], animated: false)
    presentPanModal(navigationController)
```

Hope this help.

**Comment by Pato-Salazar** [comment]
Same issue here.

**Comment by hewang1016729378** [comment]
遇到了同样的问题

**Comment by Ashok28** [comment]
Can confirm, that workaround provided by @yavl works. Moving all layout configuration from constructor to viewDidLoad (or other lifecycle methods) fixes the issue.

**Comment by Pato-Salazar** [comment]
Does not work for me. Maybe because I have a collectionView in the modal that is being presented??.. Is this library abandoned?...

**Comment by yavl** [comment]
> Does not work for me. Maybe because I have a collectionView in the modal that is being presented??.. Is this library abandoned?...

if you are presenting UICollectionViewController, try making minimal reproducible example with UIViewController + UICollectionView on it and see if that helps.

**Comment by oscarcv** [comment]
From the tests I have performed the problem appears in the `addRoundedCorners` function of the `PanModalPresentationController` class.

```swfit
 func addRoundedCorners(to view: UIView) {
        let radius = presentable?.cornerRadius ?? 0
        let path = UIBezierPath(roundedRect: view.bounds, // Bounds value is .zero
                                byRoundingCorners: [.topLeft, .topRight],
                                cornerRadii: CGSize(width: radius, height: radius))

        // Draw around the drag indicator view, if displayed
        if presentable?.showDragIndicator == true {
            let indicatorLeftEdgeXPos = view.bounds.width/2.0 - Constants.dragIndicatorSize.width/2.0
            drawAroundDragIndicator(currentPath: path, indicatorLeftEdgeXPos: indicatorLeftEdgeXPos)
        }

        // Set path as a mask to display optional drag indicator view & rounded corners
        let mask = CAShapeLayer()
        mask.path = path.cgPath
        view.layer.mask = mask

        //Improve performance by rasterizing the layer
        view.layer.shouldRasterize = true
        view.layer.rasterizationScale = UIScreen.main.scale
    }

```

Because when creating the `panContainerView` object, `containerView` is nil

```swfit
    private lazy var panContainerView: PanContainerView = {
        let frame = containerView?.frame ?? .zero // ContainerView is nill
        return PanContainerView(presentedView: presentedViewController.view, frame: frame)
    }()
```

To solve the problem, we add in line `179`, to the frame calculation if it is zero

```swift 

    override public func presentationTransitionWillBegin() {

        guard let containerView = containerView
            else { return }
        // Fix bug issue
        if self.panContainerView.frame == .zero {
            self.adjustPresentedViewFrame()
        }
    ...
```


I hope it will be of help

**Comment by Pato-Salazar** [comment]
@oscarcv this actually works for me as well. Nicely done.  Can you make a PR for this?

**Comment by oscarcv** [comment]
> @oscarcv this actually works for me as well. Nicely done. Are you cool if I create a PR with your solution?

https://github.com/slackhq/PanModal/pull/204

**Comment by Pato-Salazar** [comment]
@oscarcv I tagged the library owners.. if we get lucky they might merge this...thanks again

**Comment by abmomen** [comment]
Got same issue, on iOS 17.1.1, 14 pro max

In our case it showing up for first time and after that then only dark screen, if we try to open modal multiple times

**Comment by springfk** [comment]
any other solution?

**Comment by springfk** [comment]
for iPad it has an issue with your fix @oscarcv, any idea?

**Comment by oscarcv** [comment]
> for iPad it has an issue with your fix @oscarcv, any idea?

If you provide me with a test app to reproduce the problem, I try to review it.

**Comment by Van-Lucky** [comment]
fix problem with iOS 17.1 
![Uploading 2023-12-16 08.18.26.jpg…]()

**Comment by ppave** [comment]
Faced the same issue but with using PanModal for UIPageViewController.
Created a PR [here](https://github.com/slackhq/PanModal/pull/208/files), tested on iPhone and iPad.
Please test on your side if you can. Thanks
